routes/users.js

```diff
@@ -2,11 +2,56 @@ const express = require("express");
 const router = express.Router();
 const userService = require("../services/user_service");
 const rolesService = require("../services/roles_service");
+const applService = require("../services/applications_service");
+const admin = require("firebase-admin");
 const {verifyOwnership, OWNED_ENTITIES} = require("../middleware/authorization");
 const { isValidIntegerId, respondWithError, isUniqueConstraintViolation, isNotNullConstraintViolation } = require("./helpers");
 
 router.use(express.json());
 
+// POST /users/login
+router.post("/users/login", async (req, res) => {
+    const token = req.headers['token'];
+    if (!token) {
+        return respondWithError(res, 401, "No token provided");
+    }
+    try {
+        //verifying the firebase token
+        const decodedToken = await admin.auth().verifyIdToken(token);
+        const email = decodedToken.email;
+        const uid = decodedToken.uid;
+        //if email is verified
+        if (!decodedToken.email_verified) {
+            return respondWithError(res, 403, "Email not verified");
+        }       
+        // check if application exists  
+        const application = await applService.getApplicationByEmail(email);  
```

Comment:
I'm confused, don't we need to use [application] here? 🤔

Reply:
apparently that's how .first() works in the query. returning(*) on insert works another way 🙃

Reply:
we will need to get consistent at some point... maybe not now. but at least make all required transformations inside repositories would be nice, so we don't have to keep in mind where do we need to add the brackets and where we don't 🤔